File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
public class Solution {
    public int[] solution(int n, int m) {
        int gcd = GCD(n, m);
        int lcm = n * m / gcd;
        return new int[2] {gcd, lcm};
    }
    
    private int GCD(int a, int b)
    {
        if(b == 0) return a;
        return GCD(b, a % b);
    }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <iostream>
using namespace std;
int GCD(int a, int b);

int main(void)
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int a, b;
    cin >> a >> b;

    int gcd = GCD(a, b);
    int lcm = a * b / gcd;

    cout << gcd << '\n' << lcm;
}


int GCD(int a, int b)
{
    int temp = 0;
    while (b != 0)
    {
        temp = b;
        b = a % b;
        a = temp;
    }

    return a;
}